In [35]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [36]:
dataset = pd.read_csv("/workspaces/flexpower-task/Data/cleaned_file.csv")

# Task 2.1

How much Wind/PV (Solar) Power was forecasted to produced in German in 2021 [in MWh] on Day Ahead (da) and on Intraday (id). Hint: Be careful: you have values in MW on a quarter hourly basis, think how this translates into hourly values.

Each row represents data in a 15-minute interval, so we would have to multiply each of the entries by 0.25 to get the hourly values.

In [37]:
# Wind Power (Day-Ahead Forecast)
dataset["Wind_DA_MWh"] = dataset["Wind Day Ahead Forecast [in MW]"] * 0.25
wind_dayahead = dataset["Wind_DA_MWh"].sum()

# Wind Power (Intraday Forecast)
dataset["Wind_ID_MWh"] = dataset["Wind Intraday Forecast [in MW]"] * 0.25
wind_intraday = dataset["Wind_ID_MWh"].sum()

# PV (Solar) Power (Day-Ahead Forecast)
dataset["PV_DA_MWh"] = dataset["PV Day Ahead Forecast [in MW]"] * 0.25
pv_dayahead = dataset["PV_DA_MWh"].sum()

# PV (Solar) Power (Intraday Forecast)
dataset["PV_ID_MWh"] = dataset["PV Intraday Forecast [in MW]"] * 0.25
pv_intraday = dataset["PV_ID_MWh"].sum()

# Printing the results

print("Wind Power (Day-Ahead Forecast) = ", wind_dayahead, "MWh")
print("Wind Power (Intraday Forecast) = ", wind_intraday, "MWh")
print("PV Power (Day-Ahead Forecast) = ", pv_dayahead, "MWh")
print("PV Power (Intraday Forecast) = ", pv_intraday, "MWh")

Wind Power (Day-Ahead Forecast) =  115395406.25 MWh
Wind Power (Intraday Forecast) =  113478034.5 MWh
PV Power (Day-Ahead Forecast) =  46094572.75 MWh
PV Power (Intraday Forecast) =  46689891.5 MWh


The total forecasted Wind and PV power for 2021 in Germany are as follows:

- Wind Power (Day-Ahead Forecast) =  115,395,406.25 MWh
- Wind Power (Intraday Forecast) =  113,478,034.5 MWh
- PV Power (Day-Ahead Forecast) =  46,094,572.75 MWh
- PV Power (Intraday Forecast) =  46,689,891.5 MWh

# Task 2.2

Show the average Wind/Solar production for 2021 over a 24h period for Intraday and Day Ahead (4 lines in one graph).

In [39]:
print(dataset.dtypes)

Unnamed: 0                                             int64
hour                                                   int64
time                                                  object
Wind Day Ahead Forecast [in MW]                        int64
Wind Intraday Forecast [in MW]                         int64
PV Day Ahead Forecast [in MW]                          int64
PV Intraday Forecast [in MW]                           int64
Day Ahead Price hourly [in EUR/MWh]                  float64
Intraday Price Price Quarter Hourly  [in EUR/MWh]    float64
Intraday Price Hourly  [in EUR/MWh]                  float64
Imbalance Price Quarter Hourly  [in EUR/MWh]         float64
Wind_DA_MWh                                          float64
Wind_ID_MWh                                          float64
PV_DA_MWh                                            float64
PV_ID_MWh                                            float64
dtype: object
